### DISCUSSION TITLE
I am working with a spinner in a button

### DISCUSSION THREAD

**Discussion by GoldenBoyGG** [community/community#71917] [Programming Help]
### Body

I am working with a spinner in a button. I can't add my spinner correctly, this is a .gif image that when loading the page the spinner appears in the whole window and it should appear when clicking on the onclick event of the button.

This is my code:

```javascript
function executeAjaxRequest() {
  $(window).load(function() {
    $(".loading").fadeOut("slow");
  });
}
.loading {
  position: fixed;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  z-index: 9999;
  background: url('../gif/loading.gif') 50% 50% no-repeat rgb(249, 249, 249);
  opacity: .8;
}
```

```html
<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>

<div class="form-group col-sm-4 col-sm-offset-4 text-center">
  <button type="button" id="navegar" class="btn btn-primary" onclick="executeAjaxRequest();">Navegar</button>
  <div class="loading"></div>
</div>
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by KernelDiego** [comment]
Add the spinner class when the executeAjaxRequest event is executed, not before. Also when you finish performing your actions don't forget to remove the spinner class.


```javascript
function executeAjaxRequest() {
  $("#boxLoading").addClass("loading")
  setTimeout(() => $("#boxLoading").removeClass("loading"), 3000);
}
```

```javascript

.loading {
  position: fixed;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  z-index: 9999;
  background: url('https://www.tmogroup.asia/wp-content/uploads/2018/05/001gif.gif?x96783') 50% 50% no-repeat rgb(249, 249, 249);
  opacity: .8;
}
```

```html
<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>

<div class="form-group col-sm-4 col-sm-offset-4 text-center">
  <button type="button" id="navegar" class="btn btn-primary" onclick="executeAjaxRequest();">Navegar</button>
  <div id="boxLoading"></div>
</div>
```

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
